src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanExpressionCheck.java

```diff
@@ -113,9 +116,28 @@ public void visitToken(DetailAST ast) {
             case TokenTypes.LAND:
                 log(parent, MSG_KEY);
                 break;
+            case TokenTypes.QUESTION:
+                final DetailAST nextSibling = ast.getNextSibling();
+                if (isBooleanLiteralType(parent.getFirstChild().getType())
+                        || nextSibling != null
+                        && isBooleanLiteralType(nextSibling.getNextSibling().getType())) {
+                    log(parent, MSG_KEY);
+                }
+                break;
             default:
                 break;
         }
     }
 
+    /**
+     * Checks if a token type is a literal true or false.
+     *
+     * @param tokenType the TokenType
+     * @return true iff tokenType is LITERAL_TRUE or LITERAL_FALSE
+     */
+    private static boolean isBooleanLiteralType(final int tokenType) {
+        final boolean isTrue = tokenType == TokenTypes.LITERAL_TRUE;
+        final boolean isFalse = tokenType == TokenTypes.LITERAL_FALSE;
+        return isTrue || isFalse;
+    }
```

Comment:
I meant not to copy-paste method from other check here, but to extract it to some common util class to reuse